tiledb/sm/cpp_api/enumeration_experimental.h

```diff
@@ -275,13 +278,16 @@ class Enumeration {
       const std::string& name,
       std::vector<std::basic_string<T>>& values,
       bool ordered = false,
-      tiledb_datatype_t type = static_cast<tiledb_datatype_t>(255)) {
+      std::optional<tiledb_datatype_t> type = std::nullopt) {
     using DataT = impl::TypeHandler<T>;
     static_assert(
         DataT::tiledb_num == 1, "Enumeration string values cannot be compound");
 
-    if (type == static_cast<tiledb_datatype_t>(255)) {
-      type = DataT::tiledb_type;
+    tiledb_datatype_t dtype;
+    if (type.has_value()) {
+      dtype = type.value();
+    } else {
+      dtype = DataT::tiledb_type;
```

Comment:
`optional::value_or` will make this a one-liner. It expresses intent more clearly, as well.

https://en.cppreference.com/w/cpp/utility/optional/value_or

Reply:
Good call. Updated.